colorizer_data/writer.py

```diff
@@ -483,6 +484,29 @@ def validate_dataset(
 
         # TODO: Add validation for other required data files
 
+        # Check that all features are unique.
+        # TODO: Use dictionary to store features during writing?
+        feature_keys = [feature["key"] for feature in self.manifest["features"]]
+        duplicate_feature_keys = get_duplicate_items(feature_keys)
+        if len(duplicate_feature_keys) > 0:
+            logging.error(
+                "All feature keys must be unique! The following duplicated feature keys were detected:"
+            )
+            logging.error("   [" + ", ".join(duplicate_feature_keys) + "]")
+            logging.error("Dataset writing will now halt.")
+            raise RuntimeError("Duplicate feature keys detected in dataset manifest.")
+
+        # Should not happen because we are using a dictionary to store backdrops :)
+        backdrop_keys = [backdrop["key"] for backdrop in self.manifest["backdrops"]]
+        duplicate_backdrop_keys = get_duplicate_items(backdrop_keys)
+        if len(duplicate_backdrop_keys) > 0:
+            logging.error(
+                "All backdrop keys must be unique! The following duplicated backdrop keys were detected:"
+            )
+            logging.error("   [" + ", ".join(duplicate_backdrop_keys) + "]")
+            logging.error("Dataset writing will now halt.")
+            raise RuntimeError("Duplicate backdrop keys detected in dataset manifest.")
+
```

Comment:
Not opposed to just removing this check outright. Backdrop keys can never be duplicated because they're stored before writing as a dictionary.

Reply:
Done, thank you for the suggestion!